### ISSUE TITLE
cli: systemd notification interacts badly with `init` workflow

### ISSUE THREAD

**Issue by bdarnell** [cockroachdb/cockroach#18938]
CockroachDB supports the [systemd readiness notification protocol](https://www.freedesktop.org/software/systemd/man/sd_notify.html), so that systemd (or any process manager supporting the same protocol) can tell when the server is ready to receive traffic. We currently define "ready to receive traffic" as being fully initialized, i.e. knowing the cluster ID. 

In the 1.0 initialization workflow, the first node (run without `--join`) bootstraps and becomes ready immediately, while the remaining nodes will not become ready for the first time until they have connected to the cluster via gossip. This works fine, but has the usual drawbacks that the `--join` flag should be added to the first node's configuration for subsequent restarts.

In the 1.1 initialization workflow, nodes do not become ready until `cockroach init` has been run. This makes it difficult to use systemd to start the nodes for the first time (especially via additional layers such as ansible) - systemd defaults to waiting for the service to become ready, which will just time out.

We have a few options here:
1. Perform the init step outside of systemd: run a node without a join flag and immediately kill it (or provide an equivalent CLI step). This leads to some duplication of logic between the script that does this and the systemd service file.
2. Move the systemd notification earlier (or, more or less equivalently, remove it completely). This unblocks the service startup process, but it means that the server is not truly ready and the first queries to it may fail. I think if we do this we should also change things so that SQL connections arriving to a non-bootstrapped node are accepted, and attempts to run queries will block until the node is ready (or a timeout has passed)
3. Make yet another tweak to the initialization process: something that can be done before the first process starts that will cause initialization to occur automatically. For example, dropping a special file into the store directory.

I'm leaning towards option 2.

Note that these issues also occur with `--background` (which uses the systemd notification protocol internally). This has been a frequent source of confusion, which we have so far addressed by recommending that `--background` only be used for single-node clusters, and multi-node clusters should use shell backgrounding with `&`. Option 2 would also fix this problem and make `--background` work more like people expect.

**Comment by bdarnell** [comment]
cc @mberhault @a-robinson

**Comment by a-robinson** [comment]
I'd second option 2, or some variation of it. We've effectively already done the same thing for our Kubernetes config by removing its pod readiness checks, since k8s wanted to wait before each cockroach process was healthy before moving on to creating the next one, which didn't work well when restarting an entire cluster.
